gwt/src/main/resources/org/pentaho/ui/database/resources/databasedialog.xul

```diff
@@ -50,14 +53,14 @@
 
 				<label id="connection-name-label"
 					value="${DatabaseDialog.label.ConnectionName}" />
-				<textbox id="connection-name-text" />
+				<textbox id="connection-name-text" pen:classname="flex-row" />
```

Comment:
`flex-row` on a text box is strange. The general changes in https://github.com/pentaho/pentaho-commons-xul/pull/250 and  https://github.com/pentaho/pentaho-commons-gwt-modules/pull/935 likely make this unnecessary.

Reply:
@dcleao I checked `GwtTextbox` and it is overriding the `layout` method.

Maybe the fix is to call the `super.layout` method and define `flex="1"` here to have `--flex-item` defined, removing the need to have `flex-row` css applied

Reply:
The `flex` attribute of a XUL component is handled by the layout of the parent component (an HBox, or VBox, or other) — no component does any local handling of this attribute, afaik.

The base implementation of the layout method, in `AbstractGwtXulComponent` is really only adequate for container components such as HBox and VBox (and this implementation should really be in `AbstractGwtXulContainer`). The first line of the method immediately returns if the component is not dynamically determined to be a container (`if ( !( this instanceof XulContainer ) ) { return; }`).

Anyway, some components which are not containers override the `layout` method to persist in the managed object's DOM element properties such as dimensions, visibility, etc. Other components may not need to do anything in this method and let the base, noop implementation to be called.

Reiterating, in the referenced proposed changes to the `GwtTextBox` class, the class `flex-item-h` class is added instead, which accomplishes some of the external aspects of being a `flex-row`, but without the internal aspects of being a `flex-row` — the CSS classes `flex-row` and `flex-column` are mainly intended to be used on actual containers of flex items, and that is not the case here. Also, the class seems to be being used to gain the `auto` flexibility that flex containers get by default. The `flex-item-h` class has overall better defaults for the `flex` and `align-self` properties of textbox/input elements, varying with whether these are inside vertical or horizontal containers.